packages/twenty-server/src/engine/core-modules/billing/commands/billing-sync-plans-data.command.ts

```diff
@@ -0,0 +1,132 @@
+import { transformStripeProductDataToProductRepositoryData } from 'src/engine/core-modules/billing/utils/transform-stripe-product-data-to-product-repository-data.util';
+@Command({
+  name: 'billing:sync-plans-data',
+  description:
+    'Fetches from stripe the plans data (meter, product and price) and upserts it into the database',
+})
+export class BillingSyncPlansDataCommand extends BaseCommandRunner {
+  constructor(
+    @InjectRepository(BillingPrice, 'core')
+    private readonly billingPriceRepository: Repository<BillingPrice>,
+    @InjectRepository(BillingProduct, 'core')
+    private readonly billingProductRepository: Repository<BillingProduct>,
+    @InjectRepository(BillingMeter, 'core')
+    private readonly billingMeterRepository: Repository<BillingMeter>,
+    private readonly stripeService: StripeService,
+  ) {
+    super();
+  }
+
+  private async upsertMetersRepositoryData(
+    meters: Stripe.Billing.Meter[],
+    options: BaseCommandOptions,
+  ) {
+    meters.map(async (meter) => {
+      try {
+        if (!options.dryRun) {
+          await this.billingMeterRepository.upsert(
+            transformStripeMeterDataToMeterRepositoryData(meter),
+            {
+              conflictPaths: ['stripeMeterId'],
+            },
+          );
+        }
+        this.logger.log(`Upserted meter: ${meter.id}`);
+      } catch (error) {
+        this.logger.error(`Error upserting meter ${meter.id}: ${error}`);
+      }
+    });
+  }
+
+  private async upsertProductRepositoryData(
+    product: Stripe.Product,
+    options: BaseCommandOptions,
+  ) {
+    try {
+      if (!options.dryRun) {
+        await this.billingProductRepository.upsert(
+          transformStripeProductDataToProductRepositoryData(product),
+          {
+            conflictPaths: ['stripeProductId'],
+          },
+        );
+      }
+      this.logger.log(`Upserted product: ${product.id}`);
+    } catch (error) {
+      this.logger.error(`Error upserting product ${product.id}: ${error}`);
+    }
+  }
+
+  override async executeBaseCommand(
+    passedParams: string[],
+    options: BaseCommandOptions,
+  ): Promise<void> {
+    const billingMeters = await this.stripeService.getAllMeters();
+
+    await this.upsertMetersRepositoryData(billingMeters, options);
+
+    const billingProducts = await this.stripeService.getAllProducts();
+
+    const billingProductsIds = await Promise.all(
+      billingProducts.map(async (product) => {
+        if (isStripeValidProductMetadata(product.metadata)) {
+          await this.upsertProductRepositoryData(product, options);
+
+          return product.id;
+        } else {
+          this.logger.log(
+            `Product: ${product.id} purposefully not inserted, invalid metadata format: ${JSON.stringify(
+              product.metadata,
+            )}`,
+          );
+        }
+      }),
+    );
+
+    const billingPrices = await Promise.all(
```

Comment:
same here, could make a util

Reply:
Actually, can't we do all the operations in the same map()? Looks like you don't need to split those

Reply:
Haven't tested this but that's the idea
```typescript

    const billingPrices = await Promise.all(
      billingProducts.map(async (product) => {
        if (!isStripeValidProductMetadata(product.metadata)) {
          this.logger.log(
            `Product: ${product.id} purposefully not inserted, invalid metadata format: ${JSON.stringify(
              product.metadata,
            )}`,
          );

          return [];
        }

        await this.upsertProductRepositoryData(product, options);

        const prices = await this.stripeService.getPricesByProductId(
          product.id,
        );

        this.logger.log(
          `${prices.length} prices found for product: ${product.id}`,
        );

        return prices;
      }),
    );
   ```